native/src/main/java/io/ballerina/stdlib/crypto/PgpEncryptionGenerator.java

```diff
@@ -0,0 +1,145 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Optional;
+
+public class PgpEncryptionGenerator {
+
+    static {
+        if (Objects.isNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME))) {
+            Security.addProvider(new BouncyCastleProvider());
+        }
+    }
+
+    private final int compressionAlgorithm;
+    private final int symmetricKeyAlgorithm;
+    private final boolean armor;
+    private final boolean withIntegrityCheck;
+    private static final int BUFFER_SIZE = 8192;
+
+    public PgpEncryptionGenerator(int compressionAlgorithm, int symmetricKeyAlgorithm, boolean armor,
+                             boolean withIntegrityCheck) {
+        this.compressionAlgorithm = compressionAlgorithm;
+        this.symmetricKeyAlgorithm = symmetricKeyAlgorithm;
+        this.armor = armor;
+        this.withIntegrityCheck = withIntegrityCheck;
+    }
+
+    public void encryptStream(OutputStream encryptOut, InputStream clearIn, long length, InputStream publicKeyIn)
+            throws IOException, PGPException {
+        PGPCompressedDataGenerator compressedDataGenerator =
+                new PGPCompressedDataGenerator(compressionAlgorithm);
+        PGPEncryptedDataGenerator pgpEncryptedDataGenerator = new PGPEncryptedDataGenerator(
+                // Configure the encrypted data generator
+                new JcePGPDataEncryptorBuilder(symmetricKeyAlgorithm)
+                        .setWithIntegrityPacket(withIntegrityCheck)
+                        .setSecureRandom(new SecureRandom())
+                        .setProvider(BouncyCastleProvider.PROVIDER_NAME)
+        );
+        // Add public key
+        pgpEncryptedDataGenerator.addMethod(new JcePublicKeyKeyEncryptionMethodGenerator(
+                getPublicKey(publicKeyIn)));
+        if (armor) {
+            encryptOut = new ArmoredOutputStream(encryptOut);
+        }
+        OutputStream cipherOutStream = pgpEncryptedDataGenerator.open(encryptOut, new byte[BUFFER_SIZE]);
+        copyAsLiteralData(compressedDataGenerator.open(cipherOutStream), clearIn, length);
+        compressedDataGenerator.close();
+        cipherOutStream.close();
+        encryptOut.close();
+    }
+
+    public Object encrypt(byte[] clearData, InputStream publicKeyIn) throws PGPException, IOException {
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(clearData);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        encryptStream(outputStream, inputStream, clearData.length, publicKeyIn);
+        return ValueCreator.createArrayValue(outputStream.toByteArray());
+    }
+
+    static PGPPublicKey getPublicKey(InputStream keyInputStream) throws IOException, PGPException {
```

Comment:
We can make this private

Reply:
I have addressed this in https://github.com/ballerina-platform/module-ballerina-crypto/pull/555/commits/78e3fb4fbeab147e5c59d3a51cfec372b405651c